src/util/graphql.js

```diff
@@ -0,0 +1,122 @@
+import 'dotenv/config';
+
+/**
+ * @param {string} query - the query to execute in Cofacts GraphQL
+ * @param {object} variables
+ * @returns {object} Result from Cofacts GraphQL
+ */
+async function graphql(query, variables = {}) {
+  const resp = await fetch(process.env.COFACTS_API_URL, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ query, variables }),
+  });
+
+  if (!resp.ok) {
+    console.log('graphql query:', JSON.stringify({ query, variables }));
+    throw new Error(`HTTP error! status: ${resp.status}`);
+  }
+
+  return resp.json();
+}
+
+// Async generator that gets a batch of replies with createdAt between `from` and `to`.
+// The generator encapsulates complex pagination logic so that the function using it can focus on
+// batch processing logic without worrying pagination.
+//
+export async function* getRepliesInBatch(from, to) {
+  // Get pageInfo outside the loop since it's expensive for rumors-api
+  const {
+    data: {
+      ListReplies: {
+        pageInfo: { lastCursor },
+      },
+    },
+  } = await graphql(
+    `
+      query ListRepliesStat($from: String!, $to: String!) {
+        ListReplies(
+          filter: { createdAt: { GT: $from, LT: $to } }
+          orderBy: { createdAt: DESC }
+        ) {
+          pageInfo {
+            firstCursor
+            lastCursor
+          }
+          totalCount
```

Comment:
We can remove `totalCount` and `firstCursor` from query if we are not using them. These fields actually maps to [separate ES queries](https://github.com/cofacts/rumors-api/blob/5dc1e5efd9f398de712f84d057d518ac7b771138/src/graphql/util.js#L235).